Add Deploy, Verify, Add Facilitator script for adding GHO token clones to testnets

AJAYK-01 opened the pull request:
#### Dependency changes
- Removed duplicate hardhat-ethers from package.json that was causing conflict during installation, will not affect the project.
- Add openzepplin contracts v4.80 (as new dependency, won't conflict with v5.0.1, as it is added under a different name.
- Add hardhat etherscan for contract verification.

####  To deploy GHO token:
- ```npx hardhat run scripts/gho/deploy-gho.ts --network <network>```

#### To verify GHO token
- ```npx hardhat run scripts/gho/verify-gho.ts --network <network>```

#### To add a GHO facilitator (populate params inside script)
- ```npx hardhat run scripts/gho/add-facilitator.ts  --network <network>```

#### Added networks:
- sepolia
- goerli
- arbisepolia
- arbigoerli (however arbiscan for goerli is not working, hence contract verification script won't)

#### Deployed GHO Token addresses:
- [Sepolia](https://sepolia.etherscan.io/address/0x6e8b65fa45Bf7DAb7858c1BDfd2057b1cD194D7d)
- [Goerli](https://goerli.etherscan.io/address/0x0D33998Cba1aFfdd283768d27D7349153A55FD1a)
- [Arbitrum Sepolia](https://sepolia.arbiscan.io/address/0x10eed5bD2AEc008edc2510292D5724419ddBdD38)
- [Arbiturm Goerli](https://explorer.triangleplatform.com/arbitrum/goerli/address/0x3e510F7cfc9B4Ce53C120cfC1D1fF2Fa2E474147)

AJAYK-01 committed:
deploy and verify gho token script

Hendobox commented on contracts/contracts/gho/gho-core/src/contracts/gho/ERC20.sol:
```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MIT-only
+pragma solidity ^0.8.0;
+
+import {IERC20} from "openzeppelin-contracts-gho/token/ERC20/IERC20.sol";
```
This location isn't found. Could you ensure that you are importing from the right source. Ideally, yo want to get the contract to compile using `npx hardhat compile`.

AJAYK-01 replied:
had you done an ```npm install``` ?

Hendobox replied:
No, I hadn't. It compiles now. Thanks

Hendobox commented on contracts/contracts/gho/gho-core/src/contracts/gho/GhoToken.sol:
```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.8.0;
+
+import {EnumerableSet} from "openzeppelin-contracts-gho/utils/structs/EnumerableSet.sol";
+import {AccessControl} from "openzeppelin-contracts-gho/access/AccessControl.sol";
```
Verify source location

Hendobox commented on contracts/contracts/gho/gho-core/src/contracts/gho/GhoToken.sol:
```diff
@@ -0,0 +1,160 @@
+     */
+    constructor(address admin) ERC20("Gho Token", "GHO", 18) {
+        _setupRole(DEFAULT_ADMIN_ROLE, admin);
+    }
```
assign facilitator manager role in constructor so that we can add facilitators and mint GHO

AJAYK-01 replied:
This is the original source code of GHO Token, do we want to change that?
I am right now working on a script that will add the facilitator using the addFacilitator function, which I think would be the ideal way.

Hendobox commented on contracts/contracts/gho/gho-core/src/contracts/gho/interfaces/IGhoToken.sol:
```diff
@@ -0,0 +1,144 @@
+pragma solidity ^0.8.0;
+
+import {IERC20} from "openzeppelin-contracts-gho/token/ERC20/IERC20.sol";
+import {IAccessControl} from "openzeppelin-contracts-gho/access/IAccessControl.sol";
```
Verify import location

Hendobox commented on contracts/hardhat.config.ts:
```diff
@@ -1,8 +1,46 @@
+    arbisepolia: {
+      url: "https://arb-sepolia.g.alchemy.com/v2/mjXgVMd5C1IQaZYRMOv_tCC6GqwPMRkx",
+      accounts: [PVT_KEY]
+    },
```
We should deploy on Arbitrum-Goerli instead of Arbitrum-Sepolia since the Savvy contract exists in the former

AJAYK-01 replied:
okay will update this.
Please send some arbi gorli eth to this address ```0x4A1F47a15831A5f4Cf627414BF57145B0b47de1a```
I can't get any of the faucets to work

Hendobox commented on contracts/hardhat.config.ts:
```diff
@@ -1,8 +1,46 @@
+    apiKey: {
+      sepolia: ETHERSCAN_API_KEY,
+      goerli: ETHERSCAN_API_KEY,
+      arbisepolia: ARBISCAN_API_KEY,
```
Change to Arbitrum-Goerli

Hendobox reviewed (changes requested):
The following issues regarding importation source location, contract compilation, deployment network, and facilitator manager role require fixing

AJAYK-01 committed:
add faciliator script for gho token

AJAYK-01 committed:
revoke and remove alchemy secrets, and attempt to add arbi gorli network

AJAYK-01 commented:
- Add facilitator script has been added
- Added Arbitrum Goerli deployment, however arbiscan is not working for goerli, hence verification does not work.

Updated the original comment to include these as well.

Hendobox commented:
> * Add facilitator script has been added
> * Added Arbitrum Goerli deployment, however arbiscan is not working for goerli, hence verification does not work.
> 
> Updated the original comment to include these as well.

What could be the reason? Did you update the ARBISCAN_API key and custom chain settings?

AJAYK-01 commented:
> > * Add facilitator script has been added
> > * Added Arbitrum Goerli deployment, however arbiscan is not working for goerli, hence verification does not work.
> > 
> > Updated the original comment to include these as well.
> 
> What could be the reason? Did you update the ARBISCAN_API key and custom chain settings?

I think it is deprecated, https://goerli.arbiscan.io/
Can you access this link?

Hendobox commented:
> > > * Add facilitator script has been added
> > > * Added Arbitrum Goerli deployment, however arbiscan is not working for goerli, hence verification does not work.
> > > 
> > > Updated the original comment to include these as well.
> > 
> > 
> > What could be the reason? Did you update the ARBISCAN_API key and custom chain settings?
> 
> I think it is deprecated, https://goerli.arbiscan.io/ Can you access this link?

You're right. I have just asked if we can move to Sepolia. If yes, then we're all good

AJAYK-01 commented:
> > > > * Add facilitator script has been added
> > > > * Added Arbitrum Goerli deployment, however arbiscan is not working for goerli, hence verification does not work.
> > > > 
> > > > Updated the original comment to include these as well.
> > > 
> > > 
> > > What could be the reason? Did you update the ARBISCAN_API key and custom chain settings?
> > 
> > 
> > I think it is deprecated, https://goerli.arbiscan.io/ Can you access this link?
> 
> You're right. I have just asked if we can move to Sepolia. If yes, then we're all good

Deploying, and using the GHO token on Arbitrum Goerli works either way, we would only miss out on using the explorer and the verified contract status. But yes moving to sepolia is good if possible!
